### PR TITLE
Rename some helper AC_PosControl functions

### PR DESCRIPTION
**Pull request by WickedShell**
These function names were driving me a bit crazy when going through quadplane code, I kept mentally thinking they were actually setting current speed/accel demands in places. This refactors the following function names to have max in it, to reflect what it is actually doing:

|Old Name|New Name|
|-|-|
|set_speed_z|set_max_speed_z|
|set_accel_z|set_max_accel_z|
|set_speed_xy|set_max_speed_xy|
|set_accel_xy|set_max_accel_xy|

Miscellaneous other cleanups in this:
- Removed unneeded initilization in AC_PosControl (just the low hanging fruit, I didn't look into what any of the defined values are)
- Some consistency comment updates on the various setters. Mostly to reflect that if the set actually preformed any action the leash was automatically recalculated immediately, rather then upon the next time update_ is called. Also reflects the fact that set_speed_z can accept a negative or positive z speed, I didn't make the change, but copter spends effort making sure a negative number is passed in here, while it doesn't matter because we prompltly just do `-fabsf(speed_down)` inside the function, and the header declares that positve or negative values are accepted. Easy change in the future, but it's for a tiny win.
- Quadplane should be trying to set the failsafes on all motor channels not just the first 8. This is a valid assumption with the current boards we have, but with future board designs and changes this may actually be useful to call across all supported channels.

### CODE REVIEW

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 1]
```diff
@@ -266,8 +258,8 @@ void AC_PosControl::set_speed_z(float speed_down, float speed_up)
     }
 }
 
-/// set_accel_z - set vertical acceleration in cm/s/s
-void AC_PosControl::set_accel_z(float accel_cmss)
+/// set_max_accel_z - set vertical acceleration in cm/s/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]: set_max_accel_z - set maximum vertical acceleration in cm/s/s

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 2]
```diff
@@ -640,8 +632,8 @@ void AC_PosControl::run_z_controller()
 /// lateral position controller
 ///
 
-/// set_accel_xy - set horizontal acceleration in cm/s/s
-void AC_PosControl::set_accel_xy(float accel_cmss)
+/// set_max_accel_xy - set horizontal acceleration in cm/s/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]: set_max_accel_xy - set maximum horizontal acceleration in cm/s/s

File: libraries/AC_AttitudeControl/AC_PosControl.cpp [hunk 3]
```diff
@@ -650,8 +642,8 @@ void AC_PosControl::set_accel_xy(float accel_cmss)
     }
 }
 
-/// set_speed_xy - set horizontal speed maximum in cm/s
-void AC_PosControl::set_speed_xy(float speed_cms)
+/// set_max_speed_xy - set horizontal speed maximum in cm/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp]: set_max_speed_xy - set maximum horizontal speed in cm/s

File: libraries/AC_AttitudeControl/AC_PosControl.h [hunk 4]
```diff
@@ -75,9 +75,9 @@ class AC_PosControl
     /// get_vel_target_z - returns current vertical speed in cm/s
     float get_vel_target_z() const { return _vel_target.z; }
 
-    /// set_accel_z - set vertical acceleration in cm/s/s
-    ///     leash length will be recalculated the next time update_z_controller() is called
-    void set_accel_z(float accel_cmss);
+    /// set_max_accel_z - set vertical acceleration in cm/s/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.h]: set maximum vertical acceleration in cm/s/s

File: libraries/AC_AttitudeControl/AC_PosControl.h [hunk 5]
```diff
@@ -165,14 +165,14 @@ class AC_PosControl
     ///     this does not update the xy target
     void init_xy_controller();
 
-    /// set_accel_xy - set horizontal acceleration in cm/s/s
-    ///     leash length will be recalculated the next time update_xy_controller() is called
-    void set_accel_xy(float accel_cmss);
+    /// set_max_accel_xy - set horizontal acceleration in cm/s/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.h]: set_max_accel_xy - set maximum horizontal acceleration in cm/s/s

File: libraries/AC_AttitudeControl/AC_PosControl.h [hunk 6]
```diff
@@ -165,14 +165,14 @@ class AC_PosControl
...
+    ///     leash length will be recalculated
+    void set_max_accel_xy(float accel_cmss);
     float get_accel_xy() const { return _accel_cms; }
 
-    /// set_speed_xy - set horizontal speed maximum in cm/s
-    ///     leash length will be recalculated the next time update_xy_controller() is called
-    void set_speed_xy(float speed_cms);
+    /// set_max_speed_xy - set horizontal speed maximum in cm/s
```
Comment (amilcarlucas) [inline libraries/AC_AttitudeControl/AC_PosControl.h]: set_max_speed_xy - set maximum horizontal speed in cm/s

**Comment by WickedShell** [comment]
@OXINARF getters have also been renamed. (I had done the setters because I was tripping over them, but you are correct that we should keep the getters/setters consistent, and the getter function descriptions had the same issues)

**Comment by WickedShell** [comment]
Merging, I got a review and approval from @tridge on mumble, and outside of removing the change to plane failsafe setup (because it introduced a bug/needs more thought on reworking in the helpers).

**Comment by rmackay9** [comment]
As a side note, normally I prefer to put the max at the end of the function or parameter name.  So I probably would have renamed it to set_accel_z_max().

Still, it's OK and it's an improvement.

**Pull request merged**